web/src/pages/Environments/EnvList.tsx

```diff
@@ -0,0 +1,40 @@
+import InfiniteScroll from 'components/InfiniteScroll';
+import useInfiniteScroll from 'hooks/useInfiniteScroll';
+import {IEnvironment, useGetEnvListQuery} from '../../redux/apis/TraceTest.api';
+import {EnvironmentCard} from './EnvironmentCard';
+import * as S from './Envs.styled';
+import NoResults from './NoResults';
+
+interface IProps {
+  query: string;
+  openDialog: (mode: boolean) => void;
+  setEnvironment: (mode: IEnvironment) => void;
+}
+
+const EnvList = ({query, openDialog, setEnvironment}: IProps) => {
+  const {list, isLoading, loadMore, hasMore} = useInfiniteScroll<IEnvironment, {query: string}>(useGetEnvListQuery, {
```

Comment:
We are moving away from the useInfiniteScroll, you cal talk to @jorgeepc to see how to implement the new pagination logic.

Reply:
✅